Made --no-metadata flag working

patrickbussmann opened the pull request:
There is a `--no-metadata` flag described in readme which was not implemented yet.
With that PR its working.

siprbaum commented:
Sorry, I completely missed this PR, will have a look. Please ping me if you don't hear anything back within a week!

patrickbussmann commented:
@siprbaum *ping*

pmiossec commented:
I don't exactly know what motivate the move but in fact the option was removed on purpose. See [this commit](https://github.com/git-tfs/git-tfs/commit/4c2ab71e75f896da62f734a05588446064a05f75) of PR #283

I don't know if it's a good idea to implement it again as it is mandatory for git-tfs to work.
If you are exporting your history and after hours, for whatever reason it fails, without it, you are unable to continue and have to start again.

And that's not like this little text at the bottom of a commit is annoying...
It's even good to keep the link in case for future reference....

I forecast that some users will find a good idea to use it and in the end they will be beaten by this bad decision...

What is real the purpose you try to achieve by restoring this feature?

My point of view is more that we should clean the documentation of the remaining occurrence.

patrickbussmann commented:
I want only a move "forward" - away from TFS to Git.
So I dont care about this tfs-id in the comment. Its only disturbing. Useless informations for the future. (ok the people only knows that this is a commit imported from TFS)

Then I searched and found: "Ah, in the documentation there is already this flag mentioned".
By the way: SVN -> GIT also has the same flag.
So I tried to use it ... Nothing happens. Bad.

So I implemented it. Now I'm happy with that & documentation is correct.

If you dont like this feature then update the documentation and stick to it.
But SVN to GIT has it, too: https://git-scm.com/docs/git-svn

And I did a whole SVN to GIT Migration in 2016 with 100+ repositories for a big company.
Even with --no-metadata.
And they working happy since that time with it.

Its not needed to know this ID because the TFS will shutdown few months after success migration.
So the only thing you know is: This is an commit imported by TFS.
Nothing more, nothing less. Because TFS is offline, removed and destroyed you can not do anything with that ID.

Its a feature like the others. You must not use it - but if you want you can.
Few lines of code. Nothing more, nothing less. @pmiossec

pmiossec commented:
> By the way: SVN -> GIT also has the same flag.

I didn't know that git-svn has something similar but reading quickly the documentation:
1. it is not recommended (as of git-tfs)
2. git-svn store also the mapping in some hidden files so is able to work without it whereas git-tfs can't. So for git-tfs it would be **HIGHLY** not recommended.

> Few lines of code. Nothing more, nothing less

So even if it's only some little lines to maintain, it's probably more time spent on user support because someone used it without really understanding the consequences and come here to request help on fixing it's issue.

I think that I have implemented long ago reading the metadata from a commit git note (not sure and should be verified).
If it's the case, I think that such feature could be accepted only if in the same time this flag write the metadata in a git note that git-tfs is able to read.
And as git notes are not pushed by default, you will have the exact same output for you.

>So I dont care about this tfs-id in the comment. 

For me, you have a false feeling of not needing it... And even if it's perhaps true for you, I'm pretty sure it won't for the majority of users...
I understand that the more the time pass, the less the information is important but as it is a good practice to keep the TFVC history some months ins case the migration has not been perfect (I know, I maintain git-tfs during years), it could save your life to keep the metadata to be able to compare the changes. And I'm pretty sure that nearly 0% of git-tfs users knows that when doing there migration and finding this option.

>Its only disturbing.

:dizzy_face: A little text at the bottom of a commit is disturbing!?!
You don't need much to be disturbed....
In 2 months, you will probably nearly never read these commit messages.
I work in a migrated repository and I don't remember the last time I saw one. In fact, now I know because due to our discussion, I just search of one 😁So it's maybe just a matter of being patient some weeks....
